docs/content/develop/test.md

```diff
@@ -0,0 +1,154 @@
+2. Write the Terraform configuration for your test. This will need to include all of the required dependencies. For example:
+   ```tf
+   resource "google_compute_subnetwork" "primary" {
+     name          = "my-subnet"
+     ip_cidr_range = "10.1.0.0/16"
+     region        = "us-central1"
+     network       = google_compute_network.network.name
+   }
+
+   resource "google_compute_network" "network" {
+     name                    = "my-network"
+     auto_create_subnetworks = false
+   }
+   ```
+3. Replace the id of the primary resource you are testing with `<%= ctx[:primary_resource_id] %>`.
+4. Replace fields that are identifiers, like `id` or `name`, with an appropriately named variable. For example, `<%= ctx[:vars]['subnetwork_name'] %>`. The resulting configuration for the above example would look like this:
+   ```tf
+   resource "google_compute_subnetwork" "<%= ctx[:primary_resource_id] %>" {
+     name          = "<%= ctx[:vars]['subnetwork_name'] %>"
+     ip_cidr_range = "10.1.0.0/16"
+     region        = "us-central1"
+     network       = google_compute_network.network.name
+   }
+
+   resource "google_compute_network" "network" {
+     name                    = "<%= ctx[:vars]['network_name'] %>"
+     auto_create_subnetworks = false
+   }
+   ```
+5. Modify the relevant `RESOURCE_NAME.yaml` file under [magic-modules/mmv1/products](https://github.com/GoogleCloudPlatform/magic-modules/tree/main/mmv1/products) to include an [`examples`](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/provider/terraform/examples.rb) block with your test. For example:
+   ```yaml
+   examples:
+     - !ruby/object:Provider::Terraform::Examples
+       name: "compute_subnetwork_basic"
+       primary_resource_id: "example"
+       vars:
+         subnetwork_name: "example-subnet"
+         network_name: "example-network"
+   ```
+
+> **Warning:** Values in `vars` must include a `-` (or `_`). They [trigger the addition of a `tf-test` prefix](https://github.com/GoogleCloudPlatform/magic-modules/blob/6858338f013f5dc57729ec037883a3594441ea62/mmv1/provider/terraform/examples.rb#L244), which the sweeper uses to clean them up after tests run.
+6. If beta-only fields are being tested:
+   - Add `provider = google-beta` to every resource in the `*.tf.erb` config file.
+   - Add `min_version: beta` to the `examples` block in `RESOURCE_NAME.yaml`.
+{{< /tab >}}
+{{< tab "Handwritten" >}}
+This section assumes you've used the [Add a resource]({{< ref "/develop/resource.md" >}}) guide to create your handwritten resource, and you have a working MMv1 config.
+
+> **Note:** If not, you can create one now, or skip this guide and construct the test by hand. Writing tests by hand can sometimes be a better option if there is a similar test you can copy from.
+
+1. Add the test in MMv1. Repeat for all the standard tests you will need.
```

Comment:
This language was added here specifically to avoid needing to regenerate multiple times and do some mildly more complicated copy-and-pasting from the generated file when additions are made.

The idea of "doing this task for all tests you will need" was meant to be captured in the introductory paragraph.